lxd/daemon.go

```diff
@@ -1357,7 +1357,7 @@ func (d *Daemon) NodeRefreshTask(heartbeatData *cluster.APIHeartbeat) {
 	}
 
 	// Only refresh forkdns peers if the full state list has been generated.
-	if heartbeatData.FullStateList {
+	if heartbeatData.FullStateList && heartbeatData.Members != nil && len(heartbeatData.Members) > 0 {
```

Comment:
IIRC len(nil) returns 0, so you can save yourself part of this check

Reply:
@stgraber tested and yes it does thanks